crates/utils/src/dir.rs

```diff
@@ -0,0 +1,23 @@
+use std::{fs, io, path::PathBuf};
+
+use directories::BaseDirs;
+
+/// Creates a `.ream` directory in the system's data directory if it doesn't exist.
```

Comment:
```suggestion
/// Creates a `ream` directory in the system's data directory if it doesn't exist.
```

this comment still has `.ream`, I would prefer if we didn't create this create and just had this code in the storage crate, but if you really want to keep it, the utils crate should go in the common folder, not in the base crates directory